lib/workers/repository/process/write.ts

```diff
@@ -8,12 +8,31 @@ import { fingerprint } from '../../../util/fingerprint';
 import { branchExists, getBranchCommit } from '../../../util/git';
 import { setBranchNewCommit } from '../../../util/git/set-branch-commit';
 import { Limit, incLimitedValue, setMaxLimit } from '../../global/limits';
-import { BranchConfig, BranchResult } from '../../types';
+import {
+  BranchConfig,
+  BranchResult,
+  UpgradeFingerprintConfig,
+} from '../../types';
 import { processBranch } from '../update/branch';
+import { upgradeFingerprintFields } from './fingerprint-fields';
 import { getBranchesRemaining, getPrsRemaining } from './limits';
 
 export type WriteUpdateResult = 'done' | 'automerged';
 
+export function generateBranchFingerprintConfig(
+  branch: BranchConfig
+): UpgradeFingerprintConfig[] {
+  const res = branch.upgrades.map((upgrade) => {
+    const filtertedUpgrade = {} as any;
+    for (const field of upgradeFingerprintFields) {
+      filtertedUpgrade[field] = upgrade[field];
+    }
+    return filtertedUpgrade as UpgradeFingerprintConfig;
```

Comment:
This kind of destroys the point of types. Maybe Partial<> works better? @JamieMagee @viceice @zharinov are more experts on this than me

Reply:
i don't think we need the any and can use the type directly, as we cop only field of this config type. 🤔

Reply:
Throws error:
![image](https://user-images.githubusercontent.com/76607510/197326432-4bcae147-f3a1-4674-84be-df65c4b12bde.png)

Reply:
you need to make sure every field is also in the original upgrade config object